srcs/free.c

```diff
@@ -34,8 +34,44 @@ void free_token_list(t_token *head)
 	}
 }
 
-void free_all(char *line, t_token *token_list)
+void free_cmd_list(t_command *cmd_list)
 {
-    free_token_list(token_list);
-    free(line);
+	t_command	*current;
+	t_command	*tmp;
+	size_t		i;
+
+	if (cmd_list == NULL)
+	{
+		return ;
+	}
+	current = cmd_list;	
+	while (current != NULL)
+	{
+		tmp = current;
+		if (current->infile != NULL)
+			free(current->infile);
+		if (current->outfile != NULL)
+			free(current->outfile);		
+		if (current->heredoc != NULL)
+			free(current->heredoc);
+		i = 0;
+		while(current->args[i] != NULL)
+		{
+			free(current->args[i]);
+			i++;
+		}
+		current = current->next;
+		free(tmp);
+	}
+}
+
+//t_command *cmd_list to be included
+void free_all(char *line, t_token *token_list, t_command *cmd_list)
+{
+    if (token_list != NULL)
+		free_token_list(token_list);
+	if (cmd_list != NULL)
+		free_cmd_list(cmd_list);
+    if (line != NULL)
+		free(line);
```

Comment:
```suggestion
	free_token_list(token_list);
	free_cmd_list(cmd_list);
	free(line);
```

Same thing here, all functions are already properly safeguarded with `if (ptr == NULL) return ;` so checking is redundant.